### PR TITLE
Also output current library commits to version.cpp

### PR DESCRIPTION
**Pull request by markusbattarbee**

### CODE REVIEW

**Comment by ursg** [comment]
Nice, great idea! How does this behave in cases where any of the probed directories are not git repositories? (If they are, for example, extracted as tarballs, or, in a potential fantasy future, installed through a package system?)

**Comment by markusbattarbee** [comment]
so now it just redirects any git complaints to dev/null.

File: Makefile [hunk 1]
```diff
@@ -193,9 +193,16 @@ cleantools:
 
 # Rules for making each object file needed by the executable
 
+# Extract commits for used libraries, silencing errors of missing repositories
+COMMIT_DCCRG=$(shell cd ${subst -system,,${subst -I,,${INC_DCCRG}}} && git log -1 --pretty=format:"%H" 2>/dev/null)
+COMMIT_FSGRID=$(shell cd ${subst -system,,${subst -I,,${INC_FSGRID}}} && git log -1 --pretty=format:"%H" 2>/dev/null)
+COMMIT_VLSV=$(shell cd ${subst -system,,${subst -I,,${INC_VLSV}}} && git log -1 --pretty=format:"%H" 2>/dev/null)
+COMMIT_HASHINATOR=$(shell cd ${subst -system,,${subst -I,,${INC_HASHINATOR}}} && git log -1 --pretty=format:"%H" 2>/dev/null)
+COMMIT_PROFILE=$(shell cd ${subst -system,,${subst -I,,${INC_PROFILE}}} && git log -1 --pretty=format:"%H" 2>/dev/null)
```
Comment (ykempf) [inline Makefile:201]: So no need for a .. as that doesn't go to the top level git folder?

File: Makefile [hunk 2]
```diff
@@ -193,9 +193,16 @@ cleantools:
 
 # Rules for making each object file needed by the executable
 
+# Extract commits for used libraries, silencing errors of missing repositories
+COMMIT_DCCRG=$(shell cd ${subst -system,,${subst -I,,${INC_DCCRG}}} && git log -1 --pretty=format:"%H" 2>/dev/null)
+COMMIT_FSGRID=$(shell cd ${subst -system,,${subst -I,,${INC_FSGRID}}} && git log -1 --pretty=format:"%H" 2>/dev/null)
+COMMIT_VLSV=$(shell cd ${subst -system,,${subst -I,,${INC_VLSV}}} && git log -1 --pretty=format:"%H" 2>/dev/null)
+COMMIT_HASHINATOR=$(shell cd ${subst -system,,${subst -I,,${INC_HASHINATOR}}} && git log -1 --pretty=format:"%H" 2>/dev/null)
```
Comment (ykempf) [inline Makefile:200]: The future!

**Review by ykempf — COMMENTED** [review]
Some of the external libraries are also taken off git these days when not in modules, but this can be extended ad lib.
Would you mind pasting an example output?

File: Makefile (hunk 1 above)
Comment (markusbattarbee) [reply Makefile:201]: Indeed, git log can be run from anywhere inside the repo.

**Comment by markusbattarbee** [comment]
```
----------- Compilation --------- 
date:            Fri 27 Oct 10:06:47 EEST 2023
folder:          /users/marbat/git/vlasiator 
CMP:             OMPI_CXX='nvcc' OMPI_CXXFLAGS='' mpic++ 
CXXFLAGS:        -g -O2 -x cu -std=c++17 --extended-lambda --expt-relaxed-constexpr -gencode arch=compute_80,code=sm_80 --generate-line-info -line-info -Xcompiler=-fopenmp --prec-sqrt=true --prec-div=true --ftz=false --fmad=false -Xcompiler=-fpermissive -dc  -maxrregcount 24 -DPAPI_MEM -DIONOSPHERE_SORTED_SUMS -DINITIALIZE_ALIGNED_MALLOC_WITH_NAN -MMD -D OMPI_SKIP_MPICXX -DPROFILE -DHASHINATOR_DEBUG -DDEBUG_SPATIAL_CELL -DDEBUG_VMESH -DDEBUG_VBC -DACC_SEMILAG_PQM -DTRANS_SEMILAG_PPM -I/projappl/project_2000203/libraries_rhel8_gcccuda/phiprof_nvcc/include  -DDP -DDPF -DVEC_FALLBACK_GENERIC -DUSE_GPU -I/users/marbat/git/hashinator/ -isystem /usr/local/cuda/include -DWID=4 -DVECL=16 
FLAGS:            
INC_MPI:          
INC_ZOLTAN:      -isystem //projappl/project_2000203/libraries_rhel8_gcccuda/zoltan/include 
INC_BOOST:       -isystem //projappl/project_2000203/libraries_rhel8_gcccuda/boost/include 
INC_DCCRG:       -I/projappl/project_2002873/libraries/nvhpc/21.2/dccrg 
                 commit: cbcdbe799f017debd862b3281a534054f6abc707 
INC_FSGRID:      -I/projappl/project_2002873/libraries/nvhpc/21.2/fsgrid 
                 commit: b556505c072abb0d82de840c5af60b24dc24dd8e 
INC_VLSV:        -I/projappl/project_2000203/libraries_rhel8_gcccuda/vlsv 
                 commit: 760c9780ea67ee2f9d398c583d1f4f427ffa4e7b 
INC_HASHINATOR:  -I/users/marbat/git/hashinator/ 
                 commit: bc58f43bed3ead3d0069336dfbc7b52dddbdbe2f 
INC_PHIPROF:     -I/projappl/project_2000203/libraries_rhel8_gcccuda/phiprof_nvcc/include 
                 commit: 605a7247c85d967fe22fe079c96c817b461c92b1 

----------- git branch --------- 
```
And I've tested that when compiling the CPU branch, i.e. INF_HASHINATOR is not set, it just prints those lines without the information it wasn't able to retrieve.

**Pull request merged**